ext/IntegralsCubaExt.jl

```diff
@@ -33,24 +37,21 @@ function Integrals.__solvebp_call(prob::IntegralProblem, alg::AbstractCubaAlgori
             scale = x -> scale_x!(view(_x, :, 1:size(x, 2)), ub, lb, x)
         end
 
-        if isinplace(prob)
-            fsize = size(prob.f.integrand_prototype)[begin:(end - 1)]
-            y = similar(prob.f.integrand_prototype, fsize..., nvec)
-            ax = map(_ -> (:), fsize)
-            f = function (x, dx)
-                dy = @view(y[ax..., begin:(begin + size(dx, 2) - 1)])
-                prob.f(dy, scale(x), p)
-                dx .= reshape(dy, :, size(dx, 2)) .* vol
+        if isinplace(f)
+            ax = ntuple(_ -> (:), length(fsize))
+            _f = let y_ = similar(prototype, fsize..., nvec)
+                function (u, _y)
+                    y = @view(y_[ax..., begin:(begin + size(_y, 2) - 1)])
```

Comment:
why a view? I don't see why this isn't already the right size and just needs the reshape.